src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -162,7 +169,10 @@ private void cancel() {
                     LOGGER.log(Level.WARNING, null, x);
                     return;
                 }
-                final Throwable death = new FlowInterruptedException(Result.ABORTED, new ExceededTimeout());
+                ExceededTimeout exceededTimeout = new ExceededTimeout();
+                exceededTimeout.setNodeId(nodeId);
+                final Throwable death =
+                        new FlowInterruptedException(Result.ABORTED, exceededTimeout);
```

Comment:
I am not understanding why all the complexity. Does a simple
```suggestion
                        new FlowInterruptedException(Result.ABORTED, false, exceededTimeout);
```
not solve the issue here, without the upstream PR?

Reply:
If you do that, then I think both case 1 and case 2 from the PR description would have the same behavior. (Maybe fine, but IMO that just trades the current frustrations with case 1 to new frustrations with case 2.)